lib/galaxy/jobs/runners/__init__.py

```diff
@@ -475,8 +475,12 @@ def _finish_or_resubmit_job(self, job_state, job_stdout, job_stderr, job_id=None
             job = job_state.job_wrapper.get_job()
             exit_code = job_state.read_exit_code()
 
-            tool_stdout_path = os.path.join(job_wrapper.working_directory, "tool_stdout")
-            tool_stderr_path = os.path.join(job_wrapper.working_directory, "tool_stderr")
+            outputs_directory = os.path.join(job_wrapper.working_directory, "outputs")
+            if not os.path.exists(outputs_directory):
+                outputs_directory = job_wrapper.working_directory
+
+            tool_stdout_path = os.path.join(outputs_directory, "tool_stdout")
+            tool_stderr_path = os.path.join(outputs_directory, "tool_stderr")
```

Comment:
Is it time to do what the TODO below says?

Reply:
We could think about doing that this cycle, maybe not in this PR though?

Reply:
Not in this PR, no. Thanks @jmchilton!